src/managers/ProductManager/ProductManager.ts

```diff
@@ -0,0 +1,173 @@
+  ) => {
+    const labelElementContainer = document.createElement('p');
+    labelElementContainer.classList.add(TAG_STYLES_CLASS);
+    labelElementContainer.classList.add('product__name');
+
+    const labelElement = document.createElement('span');
+    labelElement.textContent = window.sponsoredProductConfig.tagLabel;
+    labelElementContainer.appendChild(labelElement);
+
+    if (dsaUrl) {
+      const sponsoredLabelLink = document.createElement('a');
+      sponsoredLabelLink.style.marginLeft = '4px';
+      sponsoredLabelLink.style.textDecoration = 'none';
+      sponsoredLabelLink.style.pointerEvents = 'auto';
+      sponsoredLabelLink.href = dsaUrl;
+      sponsoredLabelLink.target = '_blank';
+      sponsoredLabelLink.innerHTML = DSA_ICON;
+
+      labelElementContainer.appendChild(sponsoredLabelLink);
+    }
+
+    productElement.querySelector(selector)?.prepend(labelElementContainer);
+
+    return productElement;
+  };
+
+  private injectTagStyles = (styles: string) => {
+    const stylesElement = document.createElement('style');
+    stylesElement.innerHTML = styles;
+    this.productsContainer.appendChild(stylesElement);
+  };
+
+  public injectProduct = (products: TFormattedProduct[]) => {
+    let productsContainer = this.productsContainer;
+    const tagStyles = TAGS_STYLES[this.currentTemplate];
+
+    this.injectTagStyles(tagStyles.styles);
+
+    if (this.isSlider) {
+      // override productsContainer to be the actual container of slider items
+      productsContainer = this.productsContainer.querySelector(
+        '.slick-track',
+      ) as HTMLDivElement;
+    }
+
+    const { productsCount } =
+      window.sponsoredProductConfig[mapConfigPages(this.page)];
+
+    for (const [index, product] of products.entries()) {
+      if (index >= productsCount) break;
+
+      const productElement = document.createElement('div');
+      let productTemplateHTML = this.templateHTML;
+
+      const mappedProductValues = getProductMap(product);
+
+      for (const key in mappedProductValues) {
+        const objKey = key as keyof typeof mappedProductValues;
+
+        productTemplateHTML = productTemplateHTML.replaceAll(
+          new RegExp(key, 'g'),
+          mappedProductValues[objKey]?.toString(),
+        );
+      }
+
+      productElement.innerHTML = productTemplateHTML;
+      const preparedElement = productElement.firstChild as HTMLDivElement;
+
+      this.deleteExistingProduct(product.id);
+
+      const cleanedProductElement = this.cleanUpProductElement(
+        preparedElement,
+        mappedProductValues,
+      );
+      const taggedProductElement = this.addTagToProductElement(
+        cleanedProductElement,
+        tagStyles.selector,
+        product.dsaUrl,
+      );
+
+      taggedProductElement.classList.remove(SLIDER_CLONED_CLASS);
+      productsContainer.prepend(taggedProductElement);
+    }
+
+    if (this.isSlider) {
+      $(`${getSliderContainerSelector(this.page)} .products`).each(function () {
```

Comment:
Jak tak patrzę, to tylko dla użycia tych dwóch funkcji: each oraz slick musimy mieć w bibliotece JQuery?
Potencjalnie problematyczne na przyszłość w kotekście utrzymania